### PR TITLE
chore: [Policy Assistant] rename Cyclonus to policy-assistant

### LINKED ISSUE #170: [Policy Assistant] use internal vendoring instead of vendoring from Cyclonus
Related to #150.

See https://github.com/kubernetes-sigs/network-policy-api/pull/159#discussion_r1374029219

### LINKED ISSUE #254: [Policy Assistant] update binary names to policy-assistant instead of cyclonus
Duplicate of #171 after it was auto-closed.

Related to https://github.com/kubernetes-sigs/network-policy-api/issues/150.

See https://github.com/kubernetes-sigs/network-policy-api/pull/159#discussion_r1375079221.

### PR DESCRIPTION
**Pull request by huntergregory**
### New Binary Name

- Fixes #254

```shell
$ cd cmd/policy-assistant
$ make policy-assistant
$ ./cmd/policy-assistant/policy-assistant analyze --help
...
```

Used to be:

```shell
$ make cyclonus
$ ./cmd/cyclonus/cyclonus analyze --help
...
```

### New Module Name

- Fixes #170

```go
module sigs.k8s.io/network-policy-api/policy-assistant
```

Used to be:
```go
module github.com/mattfenwick/cyclonus
```

### Docs Update

- Adds some READMEs
- Moves outdated Cyclonus docs/examples to _archived/_ folders

#### Outputs for `go fmt`

- If there's a file that needs formatting:
  - {A62101C5-DD3B-4F0E-B860-070814974115}
- Otherwise:
  - {183626F3-44A2-4BDB-9B18-49F59C8D3C72}

### CODE REVIEW

File: .github/workflows/policy-assistant.yml [hunk 1]
```diff
@@ -42,34 +42,52 @@ jobs:
         cd cmd/policy-assistant/
         go test ./...
 
+    - name: Run Go Vet
+      run: |
+        cd cmd/policy-assistant/
+        go vet ./...
```
Comment (huntergregory) [inline .github/workflows/policy-assistant.yml:48]: This will fail with non-zero exit code if go vet detects problem

**Comment by huntergregory** [comment]
@aojea or @danwinship could you approve the changes to the Github action? And add a lgtm? Thanks!

File: cmd/policy-assistant/Makefile [hunk 2]
```diff
@@ -1,3 +1,14 @@
+# TODO: create an actual image registry + image
```
Comment (huntergregory) [inline cmd/policy-assistant/Makefile:1]: See #263

**Comment by huntergregory** [comment]
fyi @mattfenwick thinking to change `cyclonus` binary to `pola` per thoughts in #255

**Comment by danwinship** [comment]
/approve
for the workflow changes. It would really be good to get someone else to lgtm the policy agent changes...

**Comment by huntergregory** [comment]
_**Renaming binary from `cyclonus`:**_

I'd be inclined to name the CLI `pola` since it seems CLIs are usually short/abbreviated for less typing. Currently the binary is located at `cmd/policy-assistant/cmd/cyclonus/cyclonus`.

> ### New Binary Name
> Fixes https://github.com/kubernetes-sigs/network-policy-api/issues/254
> ```shell
> $ cd cmd/policy-assistant
> $ make pola
> $ ./cmd/pola/pola analyze --help
> ...
> ```
> Used to be:
> ```shell
> $ make cyclonus
> $ ./cmd/cyclonus/cyclonus analyze --help
> ...
> ```

**Comment by huntergregory** [comment]
**_Renaming go module from `cyclonus`:_**

idea for new name: `sigs.k8s.io/network-policy-api/policy-assistant`

NOTE: this would drop the _cmd/_ directory (pola is located at _cmd/policy-assistant/_)

> ### New Module Name
> 
> Fixes #170
> 
> ```go
> module sigs.k8s.io/network-policy-api/policy-assistant
> ```
> 
> Used to be:
> ```go
> module github.com/mattfenwick/cyclonus
> ```

**Comment by tssurya** [comment]
+1 to rename it to policy-assistant from cyclonus @mattfenwick any concerns?

**Comment by tssurya** [comment]
I am also OK to call the CLI pola and tags pola, its just the dumb me couldn't connect pola as the short abbreviation for policy-assistant ; but I am not sure if that is common in API/CLI usually? We usually always go with long abbreviations and add a short-name alias sorta thing for that long name... but maybe this is OK?

**Comment by huntergregory** [comment]
Thanks for the feedback @tssurya . As some context, Gateway API names [their CLI](https://github.com/kubernetes-sigs/gwctl) `gwctl`. It seems like their project name is "gwctl" as well though?

**Comment by huntergregory** [comment]
I'm convinced that `pola` isn't a great idea after Surya's feedback above and Mike's feedback offline, where he echoed how it wasn't immediately clear what `pola` meant. Regarding abbreviations / more typing, Mike also shared how devs frequently write their own aliases like `k` for `kubectl`.

I feel like `policy-assistant` is a decent name for the static analysis tool. And the project is already called policy assistant, so it makes for a nicer user experience if the project name is the same as the binary name. Updating the PR now to replace `pola` with `policy-assistant`

**Comment by tssurya** [comment]
Yea policy-assisstant ++ 
pola wasn't intuitive enough
Thanks @huntergregory

**Pull request merged**